snapcraft/cli/_options.py

```diff
@@ -362,9 +362,14 @@ def get_project(*, is_managed_host: bool = False, **kwargs):
         snapcraft_yaml_file_path=snapcraft_yaml_file_path,
         is_managed_host=is_managed_host,
     )
-    # TODO: this should be automatic on get_project().
-    # This is not the complete meta parsed by the project loader.
-    project._snap_meta = Snap.from_dict(project.info.get_raw_snapcraft())
+
+    # Validate yaml info from schema prior to consumption.
+    if project.info is not None and not project.is_legacy_project():
+        project.info.validate_raw_snapcraft()
```

Comment:
Is this also validating against plugin schemas?

Reply:
it does not. Plugins are evaluated on load.